src/ResourceManager.cpp

```diff
@@ -126,12 +126,12 @@ bool ResourceManager::IsDuplicateFilename(std::vector<std::string>& currentFilen
 	return false;
 }
 
-std::string ResourceManager::FindContainingArchive(const std::string& filename)
+std::string ResourceManager::FindContainingArchivePath(const std::string& filename)
 {
 	for (const auto& archiveFile : ArchiveFiles)
 	{
 		if (archiveFile->Contains(filename)) {
-			return XFile::GetFilename(archiveFile->GetVolumeFilename());
+			return XFile::ReplaceFilename(resourceRootDir, XFile::GetFilename(archiveFile->GetVolumeFilename()));
```

Comment:
I don't believe XFile is capable of appending two directories together. Someone would have to create a new function on XFile or directly include C++17 filesystem in ResourceManager to accomplish.

Reply:
We need to rename archiveFile->GetVolumeFilename to GetFilename. That must have been an oversight previously on my part when renaming.

GetVolumeFilename will return whatever path was used to open the archive. It could be relative or absolute.

Reply:
You're right, it does just return whatever it was opened with. The `ResourceManager` uses `XFile::GetFilesFromDirectory` to scan for VOL and CLM files, and then passes those names directly to the `VolFile` or `ClmFile` constructors. Whether paths are relative or absolute depends on how `GetFilesFromDirectory` was implemented.

The `GetFilesFromDirectory` method was implemented with [`directory_iterator`](https://en.cppreference.com/w/cpp/filesystem/directory_iterator), though the documentation is unclear if the `directory_iterator` returns absolute or relative paths. If they are relative it's also not clear what exactly they are relative to. The example "possible output" included the relative path passed in to the `directory_iterator` as part of the output.

----

You're right, that name probably should be simplified. We might want to do the rename in it's own branch though. That change will need to touch a few files, including some in OP2Archive.